distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/Manager.java

```diff
@@ -86,6 +99,49 @@ public void start() throws Exception {
 
         leaderLatch.start();
         workersAliveChildrenCache.start();
+        startWorkPoolCleaningTask();
+    }
+
+    private void startWorkPoolCleaningTask() {
+            workPoolCleaningReschedulableScheduler.schedule(
+                    DynamicProperty.delegated(() -> Schedule.withDelay(workPoolCleanPeriodMs.get())),
+                    () -> {
+                        try {
+                            TransactionalClient.tryCommit(
+                                    curatorFramework,
+                                    CLEAN_WORK_POOL_RETRIES_COUNT,
+                                    transaction -> {
+
+                                        String workPoolVersion = paths.availableWorkPoolVersion();
+                                        int version = curatorFramework.checkExists().forPath(workPoolVersion).getVersion();
+                                        transaction.checkPathWithVersion(workPoolVersion, version);
```

Comment:
This pattern happening is several places. 
Let's make a method with readable name
```
getVersion,
transaction.checkPathWithVersion(path, version); 
transaction.setData(path, new byte[]{});
```
Take a look into ru.fix.distributed.job.manager.Worker#checkAndUpdateVersion
can be moved in to TransactionalClient as a method or 
or separate TransactionalVersionChecker class withing DJM

Reply:
pull-request created https://github.com/ru-fix/jfix-zookeeper/pull/6

Reply:
fixed